### PR TITLE
Use localhost:4000 as the default pages host in development

### PR DESCRIPTION
**Pull request by benbalter**
Fixes https://github.com/jekyll/github-metadata/issues/47.

### CODE REVIEW

**Comment by parkr** [comment]
@benbalter I thought `PAGES_ENV` and `JEKYLL_ENV` were different things? Why would we conflate them?

**Comment by benbalter** [comment]
>  I thought PAGES_ENV and JEKYLL_ENV were different things? Why would we conflate them?

Explained my understanding in https://github.com/jekyll/github-metadata/pull/49#issuecomment-209963969. Would love to hear your thoughts there, as I suspect the two discussions are really one.

File: spec/pages_spec.rb [hunk 1]
```diff
@@ -33,21 +33,38 @@
       end
     end
 
+    it "picks up on JEKYLL_ENV" do
+      with_env "PAGES_ENV", "" do
+        with_env("JEKYLL_ENV", "halp") do
```
Comment (parkr) [inline spec/pages_spec.rb]: Consistent parentheses please 😄

**Comment by benbalter** [comment]
@parkr now that tests pass, how does this look to you?

**Comment by parkr** [comment]
LGTM

**Comment by benbalter** [comment]
LGTM.

**Comment by benbalter** [comment]
@jekyllbot: merge +major

**Pull request merged**